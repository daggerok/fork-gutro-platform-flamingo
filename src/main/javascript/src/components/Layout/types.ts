export type HeaderProps = {
    children: React.ReactNode;
};

export type LayoutProps = {
    children: React.ReactNode;
};
