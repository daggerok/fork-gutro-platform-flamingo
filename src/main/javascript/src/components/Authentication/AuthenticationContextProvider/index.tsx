import React, { createContext, useState } from 'react';

type AuthenticationContextProps = { 
  authenticated: boolean | null;
  setAuthenticated: (authenticated: boolean | null) => void;
};

export const AuthenticationContext = createContext<AuthenticationContextProps>({
  authenticated: null,
  setAuthenticated: () => {}, // eslint-disable-line @typescript-eslint/no-empty-function
});

type AuthenticationContextProviderProps = {
  children: React.ReactNode;
}

const AuthenticationContextProvider: React.FC<AuthenticationContextProviderProps> = ({ children }: AuthenticationContextProviderProps) => {
  const [authenticated, setAuthenticated] = useState<boolean | null>(null);

  return (
    <AuthenticationContext.Provider value={{ authenticated, setAuthenticated }}>
      {children}
    </AuthenticationContext.Provider>
  );
};

export default AuthenticationContextProvider;
