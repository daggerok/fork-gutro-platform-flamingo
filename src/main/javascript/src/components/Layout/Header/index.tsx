import React from 'react';
import { Layout } from 'antd';
const { Header: AntdHeader } = Layout;
import EnvironmentFlag from '~/components/Common/EnvironmentFlag';
import { HeaderProps } from '../types';
import styles from './Header.module.scss';
import logo from './logo.png';

const Header: React.FC<HeaderProps> = ({ children }: HeaderProps) => (
  <AntdHeader 
    data-testid="header"
    className={styles.header}
  >
    <img
      src={logo}
      data-testid="header-logo"
      className={styles.logo}
    />
    <span 
      data-testid="header-title"
      className={styles.title}
    >
      Flamingo
    </span>
    <div 
      data-testid="header-flag-container"
      className={styles.flagContainer}
    >
      <EnvironmentFlag />
    </div>
    { children }
  </AntdHeader>
);

export default Header;
