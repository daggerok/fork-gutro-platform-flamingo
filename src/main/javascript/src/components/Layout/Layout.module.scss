.layout {
  width: 100vw;
  height: 100vh;
  position: relative;
}

.menu {
  display: inline-block;
}

.divider {
  flex-grow: 1;
}

.logoutButton {
  display: inline-block;
}

.toTheLeft {
  float:right;
}

.userName {
  color:rgba(255, 255, 255, 0.65);
}

.userName  {
  color:rgba(255, 255, 255, 0.65);
  margin-right: 2em;

  & .userIcon {
    color:rgba(255, 255, 255, 0.65);
    margin: 10px 10px;
  }
}

.content {
  padding: 20px;
  overflow: auto;
}
