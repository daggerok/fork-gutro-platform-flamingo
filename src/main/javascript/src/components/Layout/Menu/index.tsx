import React, { useState, useEffect } from 'react';
import idx from 'idx';
// import Link from 'next/link';

import { Menu as AntdMenu } from 'antd';

const menuItems = [
  {
    title: '.CSV sending',
    url: '/csv-upload',
  },
];

const getSelectedMenuItemUrl = (): string | null => {
  const currentPathName = idx(global, _ => _.window.location.pathname);
  const selectedMenuItem = menuItems.find(menuItem => menuItem.url === currentPathName);

  if (selectedMenuItem) {
    return selectedMenuItem.url;
  }

  return null;
};

const Menu: React.FC = () => {
  const [ selectedKey, setSelectedKey ] = useState<string | null>(null);

  useEffect(() => {
    const newSelectedKey = getSelectedMenuItemUrl();
    if (newSelectedKey !== selectedKey) {
      setSelectedKey(newSelectedKey);
    }
  });
  
  return (
    <AntdMenu
      theme="dark"
      mode="horizontal"
      selectedKeys={[selectedKey || '']}
    >
      { menuItems.map(menuItem => (
        <AntdMenu.Item key={menuItem.url}>
          {/* <Link href={menuItem.url}>
            <a>{ menuItem.title }</a>
          </Link>   */}
        </AntdMenu.Item>
      ))}
    </AntdMenu>
  );
};

export default Menu;
