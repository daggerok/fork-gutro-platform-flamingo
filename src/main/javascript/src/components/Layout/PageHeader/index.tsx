import React from 'react';
import { Row, Col, Button } from 'antd';
import { ArrowLeftOutlined } from '@ant-design/icons';
import { useHistory } from 'react-router-dom';

import { PageHeaderProps } from './types';

import styles from './PageHeader.module.scss';

const PageHeader: React.FC<PageHeaderProps> = ({ 
  title,
  subtitle,
  backgroundHex = '#777777',
  backPath,
}: PageHeaderProps) => {

  const history = useHistory();

  const goBack = (): void => {
    if (backPath) { history.push(backPath); }
  };

  return (
    <header 
      className={styles.pageHeader}
      style={{backgroundColor: backgroundHex}}
    >
      <Row align="middle">
        { backPath &&  
          <Col span={1}>  
            <Button 
              type="default" 
              ghost
              shape="circle" 
              icon={<ArrowLeftOutlined />}
              onClick={goBack}
            />
          </Col>
        }
        <Col span={backPath? 21 : 24}><h1>{title}</h1></Col>
      </Row>
      { subtitle && 
        <Row>
          <Col 
            offset={backPath ? 1 : 0}
            span={24}
          >
            <p className={styles.subtitle}>{ subtitle }</p>
          </Col>
        </Row>
      }
    </header>  
  );
};

export default PageHeader;
