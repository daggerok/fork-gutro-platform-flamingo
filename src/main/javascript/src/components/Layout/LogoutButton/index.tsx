import React, { useContext } from 'react';
import { Button } from 'antd';

import { AuthenticationContext } from '~/components/Authentication/AuthenticationContextProvider';
import { logoutPlayer } from '~/api/auth';

const LogoutButton: React.FC = () => {
  const { setAuthenticated } = useContext(AuthenticationContext);

  const handleLogoutClick = (): void => {
    setAuthenticated(false);
    logoutPlayer();
  };

  return (
    <Button 
      key="logout-button"
      data-testid="logout-button-wrapper"
      onClick={handleLogoutClick}
      ghost
    >
      Log Out
    </Button>
  );
};

export default LogoutButton;
