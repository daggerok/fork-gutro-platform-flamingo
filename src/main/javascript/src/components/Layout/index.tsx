import React, { useContext } from 'react';
import { Layout as AntdLayout } from 'antd';
const { Content } = AntdLayout;
import { UserOutlined } from '@ant-design/icons';

import Header from './Header';
import Menu from './Menu';
import { AuthenticationContext } from '~/components/Authentication/AuthenticationContextProvider';

import { LayoutProps } from './types';
import LogoutButton from './LogoutButton';
import styles from './Layout.module.scss';

const Layout: React.FC<LayoutProps> = ({ children }: LayoutProps) => {
  const { authenticated, user } = useContext(AuthenticationContext);

  return (
    <AntdLayout 
      className={styles.layout} 
      data-testid="layout-element"
    >
      <Header>
        { authenticated && (
          <>
            <div className={styles.menu}>
              <Menu />
            </div>
            <div className={styles.divider} />
            { user && 
              <div className={styles.userName}>
                <UserOutlined className={styles.userIcon} /> 
                { user?.firstname }
              </div>
            }
            <div className={styles.logoutButton}>
              <LogoutButton />
            </div>
          </>
        )}
      </Header>
      <Content className={styles.content}>
        { children }
      </Content>
    </AntdLayout>
  );
};

export default Layout;
