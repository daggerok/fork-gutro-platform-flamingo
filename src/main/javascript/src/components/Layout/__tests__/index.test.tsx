import React from 'react';
import { act } from '@testing-library/react-hooks';
import { render, screen } from '@testing-library/react';

import Layout from '..';

import { AuthenticationContext } from '~/components/Authentication/AuthenticationContextProvider';

const CONTEXT_PROPS = {
  value: {
    authenticated: false,
    setAuthenticated: (x:any) => x,
    user: {
      email: 'Test',
      username: 'Test',
      firstname: 'Test',
      lastname: 'Test',
      userId: 'Test',
      roles: [],
    },
    setUser: () => { }, // eslint-disable-line @typescript-eslint/no-empty-function
  }
}

const setAuthenticatedContextProp = (authenticated: boolean) => {
  return {
    ...CONTEXT_PROPS,
    value: {
      ...CONTEXT_PROPS.value,
      authenticated
    }
  };
};

describe('<Layout />', () => {

  describe('when authenticated is true', () => {

    it('matches snapshot when authenticated is true', () => {
      act(() => {
        render(
          <AuthenticationContext.Provider {...setAuthenticatedContextProp(true)}>
              <Layout children/>
          </AuthenticationContext.Provider>
        );
      });

      expect(screen.getByTestId('layout-element')).toMatchSnapshot();
    });

    it('renders logout button', () => {
      act(() => {
        render(
          <AuthenticationContext.Provider {...setAuthenticatedContextProp(true)}>
              <Layout children/>
          </AuthenticationContext.Provider>
        );
      });

      expect(screen.getByTestId('logout-button-wrapper')).toBeInTheDocument();
    });

  });

  describe('when authenticated is false', () => {

    it('matches snapshot when authenticated is false', () => {

      act(() => {
        render(
          <AuthenticationContext.Provider {...CONTEXT_PROPS}>
              <Layout children/>
          </AuthenticationContext.Provider>
        );
      });

      expect(screen.getByTestId('layout-element')).toMatchSnapshot();
    });

  });

});