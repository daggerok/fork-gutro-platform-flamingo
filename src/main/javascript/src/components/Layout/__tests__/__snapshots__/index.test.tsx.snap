// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> when authenticated is false matches snapshot when authenticated is false 1`] = `
<section
  class="ant-layout layout"
  data-testid="layout-element"
>
  <header
    class="ant-layout-header header"
    data-testid="header"
  >
    <img
      class="logo"
      data-testid="header-logo"
      src="logo.png"
    />
    <span
      class="title"
      data-testid="header-title"
    >
      Flamingo
    </span>
    <div
      class="flagContainer"
      data-testid="header-flag-container"
    />
  </header>
  <main
    class="ant-layout-content content"
  />
</section>
`;

exports[`<Layout /> when authenticated is true matches snapshot when authenticated is true 1`] = `
<section
  class="ant-layout layout"
  data-testid="layout-element"
>
  <header
    class="ant-layout-header header"
    data-testid="header"
  >
    <img
      class="logo"
      data-testid="header-logo"
      src="logo.png"
    />
    <span
      class="title"
      data-testid="header-title"
    >
      Flamingo
    </span>
    <div
      class="flagContainer"
      data-testid="header-flag-container"
    />
    <div
      class="menu"
    >
      <ul
        class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark"
        data-menu-list="true"
        data-testid="main-menu"
        role="menu"
        style="width: 512px;"
        tabindex="0"
      />
      <div
        aria-hidden="true"
        style="display: none;"
      />
    </div>
    <div
      class="divider"
    />
    <div
      class="userName"
    >
      <span
        aria-label="user"
        class="anticon anticon-user userIcon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="user"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
          />
        </svg>
      </span>
      Test
    </div>
    <div
      class="logoutButton"
    >
      <button
        class="ant-btn ant-btn-background-ghost"
        data-testid="logout-button-wrapper"
        type="button"
      >
        <span>
          Log Out
        </span>
      </button>
    </div>
  </header>
  <main
    class="ant-layout-content content"
  />
</section>
`;
